examples/dataflow-custom-templates/go/cmd/wordcount.go

```diff
@@ -0,0 +1,87 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/apache/beam/sdks/v2/go/pkg/beam"
+	"github.com/apache/beam/sdks/v2/go/pkg/beam/io/textio"
+	"github.com/apache/beam/sdks/v2/go/pkg/beam/transforms/stats"
+	"github.com/apache/beam/sdks/v2/go/pkg/beam/x/beamx"
+	"github.com/professional-services/examples/dataflow-custom-templates/go/internal/wordcount"
```

Comment:
Can we change the package structure to avoid absolute paths since this would break when the directory structure changes.

How about following structure:
go/wordcount.go

go/transforms/{display.go, filters.go, strings.go}

Changing the import to "go/transforms" in wordcount.go

Reply:
**In a module context, Go's convention is to use absolute path imports.**

If one were to change the package name to `go/internal/wordcount` or `internal/wordcount`, the following error results:
```
package internal/wordcount is not in GOROOT
```
This is so much a convention that the IDE plugin autogenerates as such.  Changes in directory naming and structure are relevant at the go.mod containing directory.  For example, in this simple project, if one were to change the name of the `go` directory in `professional-services/examples/dataflow-custom-templates/go` to `bozotheclown`, without changing the absolute import path, everything still works.  Though, convention would make it best practice to change `github.com/professional-services/examples/dataflow-custom-templates/go/internal/wordcount` to `github.com/professional-services/examples/dataflow-custom-templates/bozotheclown/internal/wordcount`

Below are some examples of popular open source Go projects and how they handle import paths:
1. https://github.com/apache/beam/blob/master/sdks/go/examples/xlang/wordcount/wordcount.go#L34
2. https://github.com/GoogleCloudPlatform/cloud-sql-proxy/blob/main/cmd/root.go#L35-L38
3. https://github.com/GoogleCloudPlatform/terraform-validator/blob/main/cmd/list_supported_resources.go#L24